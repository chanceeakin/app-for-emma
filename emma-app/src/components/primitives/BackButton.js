import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  backButton: {
    padding: 5,
    paddingBottom: 15,
  },
})

type Props = {
  title: string,
  onPress: Function,
};

const BackButton = ({ onPress, title }: Props) => (
  <TouchableOpacity onPress={onPress} style={styles.backButton}>
    <Text>{title}</Text>
  </TouchableOpacity>
)

export default BackButton
