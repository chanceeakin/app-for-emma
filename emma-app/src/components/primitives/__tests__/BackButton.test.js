import React from 'react'
import BackButton from '../BackButton'

import renderer from 'react-test-renderer'

const props = {
  title: 'onpress',
  onPress: jest.fn(),
}

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableOpacity')
})

describe('<BackButton />', () => {
  afterAll = () => {
    jest.mockRestore()
  }
  it('renders correctly', () => {
    const tree = renderer.create(<BackButton {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
