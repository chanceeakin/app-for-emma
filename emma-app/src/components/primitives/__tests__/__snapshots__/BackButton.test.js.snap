// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "padding": 5,
      "paddingBottom": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    onpress
  </Text>
</TouchableOpacity>
`;
